Tests/RegexTests/MatchTests.swift

```diff
@@ -831,5 +831,156 @@ extension RegexTests {
+    // the character are members of the property class.
+    
+    // \w
+    matchTests(
+      #"^\w$"#,
+      (eComposed, eComposed),
+      (eDecomposed, eDecomposed))
+    // \p{Letter}
+    matchTest(#"\p{Letter}$"#, input: eComposed, match: eComposed)
+    // FIXME: \p{Letter} doesn't match a decomposed character
+    matchTest(#"\p{Letter}$"#, input: eDecomposed, match: eDecomposed,
+              xfail: true)
+    
+    // \d
+    matchTest(#"\d"#, input: "5", match: "5")
+    // FIXME: \d shouldn't match a digit composed with a non-digit character
+    matchTest(#"\d"#, input: "5\u{305}", match: nil,
+              xfail: true)
+    // \p{Number}
+    matchTest(#"\p{Number}"#, input: "5", match: "5")
+    // FIXME: \p{Number} shouldn't match a number composed with a non-number character
+    matchTest(#"\p{Number}"#, input: "5\u{305}", match: nil,
+              xfail: true)
+    
+    // \s
+    matchTest(#"\s"#, input: " ", match: " ")
+    // FIXME: \s shouldn't match a number composed with a non-number character
+    matchTest(#"\s\u{305}"#, input: " ", match: nil,
+              xfail: true)
+    // \p{Whitespace}
+    matchTest(#"\s"#, input: " ", match: " ")
+    // FIXME: \p{Whitespace} shouldn't match whitespace composed with a non-whitespace character
+    matchTest(#"\s\u{305}"#, input: " ", match: nil,
+              xfail: true)
+  }
+  
+  func testCanonicalEquivalenceCustomCharacterClass() throws {
+    // Expectation: Concatenations with custom character classes should be able
+    // to match within a grapheme cluster. That is, a regex should be able to
+    // match the scalar values that comprise a grapheme cluster in separate,
+    // or repeated, custom character classes.
+    
+    matchTests(
+      #"[áéíóú]$"#,
+      (eComposed, eComposed),
+      (eDecomposed, eDecomposed))
+
+    // FIXME: Custom char classes don't use canonical equivalence with composed characters
+    matchTest(#"e[\u{301}]$"#, input: eComposed, match: eComposed,
+              xfail: true)
+    matchTest(#"e[\u{300}-\u{320}]$"#, input: eComposed, match: eComposed,
+              xfail: true)
+    matchTest(#"[a-z][\u{300}-\u{320}]$"#, input: eComposed, match: eComposed,
+              xfail: true)
+
+    // FIXME: Custom char classes don't match decomposed characters
+    matchTest(#"e[\u{301}]$"#, input: eDecomposed, match: eDecomposed,
+              xfail: true)
+    matchTest(#"e[\u{300}-\u{320}]$"#, input: eDecomposed, match: eDecomposed,
+              xfail: true)
+    matchTest(#"[a-z][\u{300}-\u{320}]$"#, input: eDecomposed, match: eDecomposed,
+              xfail: true)
+
+    let flag = "🇰🇷"
+    matchTest(#"🇰🇷"#, input: flag, match: flag)
+    matchTest(#"[🇰🇷]"#, input: flag, match: flag)
+    matchTest(#"\u{1F1F0}\u{1F1F7}"#, input: flag, match: flag)
+    
+    // First Unicode scalar followed by CCC of regional indicators
+    matchTest(#"\u{1F1F0}[\u{1F1E6}-\u{1F1FF}]"#, input: flag, match: flag)
```

Comment:
- Test cases where you toss in a full Character or multi-scalar emoji into custom classes alongside scalar literals.